### ISSUE TITLE
Minify default `prefix`

### ISSUE THREAD

**Issue by zenflow** [tests-always-included/gulp-css2js#2]
It would be very convenient if this ...

https://github.com/tests-always-included/gulp-css2js/blob/471a939ddb6850da884d7c5e779b5bd8a9431f95/gulp-css2js.js#L205-L219

... was already minified for me, or if there was an option to use a minified version.

Then in the long example in the README, there would be no need for this:

```js
// Minify the JavaScript
    .pipe(uglify())
```

I suggest keeping the code **un**minified in Git and minifying it programatically, either (a) as a build step before publishing this package, or (b) during runtime, only once, the first time a css2js stream is created.

What do you think @tests-always-included? Happy to make a PR if you like this idea. If so, would you prefer to go route (a) or (b)? (a) would be better for the user because it would take less processing power during runtime, but would mean introducing a build step in this package.

p.s. Awesome plugin! 🥇 I am surprised more people are not using this method and this plugin.

**Comment by fidian** [comment]
So, you're thinking of replacing the `prefix` value with something like this?

    prefix: '!function(d,c){' +
        'var l=d.createElement("style"),s=l.styleSheet;' +
        'd.getElementsByTagName("head")[0].appendChild(l);' +
        'if(s&&!s.disabled)s.cssText=c;else try{l.innerHTML=c}catch(e){l.innerText=c}' +
        '}(document,"',
    suffix: '");'

**Comment by fidian** [comment]
I forgot to mention earlier that I agree that code should be unminified in git whenever it's reasonable. My hand-optimized code is better than UglifyJS's attempt at minification (barely). If you want to throw together a pull request that minifies a fragment of JS and includes it as a build step, I would love to see it. If you think we could just swap out the `prefix` and `suffix` values with what's above, I'm also game to receive that as a pull request.

**Comment by zenflow** [comment]
I would use [babel-plugin-preval](https://github.com/kentcdodds/babel-plugin-preval) to do it nice and simply. Happy side-effect of this PR will be that since there's a build step (and babel) we can upgrade this package code to ES2017

**Comment by zenflow** [comment]
I like the idea of keeping the code readable : ) 

If you don't disapprove of my approach, I will get to this in a bit!

**Comment by fidian** [comment]
If you are willing, I would like to see a pull request that uses preval to minify the code. Hold off on a full ES2017 conversion for now. After the dust settles from the first one and it gets merged, feel free to submit another pull request to convert the project to ES2017.

One thing is that I would like the build step to happen automatically, but I haven't invested enough time to find a good solution that would run a build, run tests, tag, bump the version number and submit to NPM. Any ideas on that? Note: this is not a requirement for your pull request; a manual build step is just fine and it's exactly what I would expect.

**Comment by zenflow** [comment]
> I would like the build step to happen automatically

That is my preference as well. See my comments on PR #4. It generally happens automatically, but with some limitation.

> a good solution that would run a build, run tests, tag, bump the version number and submit to NPM

Can't say I know a good solution for automating **all** of that, but at least my PR doesn't add to the manual process of making a release. I would add `npm test` to the "prepublish" package script, and try looking for some other CLI programs to also add there to cover the other things .

**Comment by fidian** [comment]
Your commits made it into master. Published 1.1.0.

**Comment by fidian** [comment]
Sorry, I forgot to explain how happy I was for your contributions. Thanks for the suggestions, the code, and for also adding your name to the author list! I appreciate your work very much.

**Comment by zenflow** [comment]
Thank you @fidian, it's really great to hear my work is so appreciated 😃 

It is my pleasure to work with you on this great package!

p.s. Sorry I forgot to do so in my PR.. if you have a chance you could remove the minifying part from the example in the README

**Comment by fidian** [comment]
Consider it done. Released 1.1.1 with the updated documentation.

**Comment by zenflow** [comment]
Hey @fidian .. today I learned `npm version` will bump the version number in package.json for you

https://docs.npmjs.com/cli/version

**Comment by zenflow** [comment]
.. and make a commit in Git and tag it

**Comment by fidian** [comment]
Wonderful. Thanks for teaching me about it.
